transport_plugins/websocket/iotile_transport_websocket/device_adapter.py

```diff
@@ -78,31 +168,539 @@ def probe_async(self, callback):
                     failure_reason: None if success is True, otherwise a reason for why we could not probe
         """
 
-        result = self.client.send_command('scan', {})
-        if not result['success']:
-            callback(self.id, False, 'Error while scanning through ws : {}'.format(result['reason']))
-        else:
-            # FIXME: async after creating WebSocketHandler2
-            for dev in result['devices']:
-                self._trigger_callback('on_scan', self.id, dev, self.get_config('expiration_time'))
-            callback(self.id, True, None)
+        if self.probe_callback is not None:
```

Comment:
There can be multiple async probes in flight at the same time from different clients so this is going to need to get enhanced at some point.  It's okay to leave for now but please file an issue so we don't forget about it.